src/app/loadout-builder/mod-utils.ts

```diff
@@ -96,17 +96,19 @@ export function assignModsToArmorSet(
     }
   }
 
-  assignAllSeasonalMods(processItems, lockedArmor2Mods.seasonal, assignments);
```

Comment:
This and the corresponding change in processUtils is what I believe is fixing the issue. I need to have a think about if this order of doing general mods and seasonal mods should have a defined precedence between the two. If there shouldn't be then I will try and figure out what the problem is as I think it will have to do with the sort order.